
 GHC 6.10 requires us to be explicit about the types of exceptions. Patch
 sent upstream.

  -- Chris Lamb <lamby@debian.org>  Sun, 08 Mar 2009 03:49:54 +0000

--- a/Bustle.hs	2009-03-05 22:29:48.000000000 +0000
+++ b/Bustle.hs	2009-03-05 22:55:37.000000000 +0000
@@ -19,7 +19,7 @@
 module Main where
 
 import Prelude hiding (catch, log)
-import Control.Exception (catch)
+import Control.Exception (catch, IOException)
 
 import Control.Arrow ((&&&), (***))
 
@@ -113,7 +113,7 @@
 
     iconName <- getDataFileName "bustle.png"
     let load x = pixbufNewFromFile x >>= windowSetIcon window
-    foldl1 (\m n -> m `catch` const n)
+    foldl1 (\m n -> m `catch` (\e -> const n (e :: IOException)))
       [ load iconName
       , load "bustle.png"
       , putStrLn "Couldn't find window icon. Oh well."
